cpp/inc/bond/core/stl_containers.h

```diff
@@ -230,12 +230,9 @@ use_map_allocator_for_keys
 
 
 template <typename T> struct
-use_map_allocator_for_keys
-<T, typename boost::enable_if<
-    std::is_same<typename std::allocator_traits<typename T::allocator_type>::template rebind_alloc<int>,
-    typename std::allocator_traits<typename element_type<T>::type::first_type::allocator_type>::template rebind_alloc<int> > >::type>
-    : std::integral_constant<bool,
-        !detail::is_default_allocator<typename T::allocator_type>::value> {};
+use_map_allocator_for_keys<T, typename boost::enable_if<
+    std::uses_allocator<typename element_type<T>::type::first_type, typename T::allocator_type> >::type>
+    : std::true_type {};
```

Comment:
It looks like with this change, we'll go down allocator paths even if the default allocator is being used. This shouldn't have an observable effect for correctly implemented collections, right?

Reply:
Yes, it should not matter if specifically `std::allocator` was used, unless there is a bug in that code.